cookbook/console/command_in_controller.rst

```diff
@@ -0,0 +1,94 @@
+
+You may have the need to execute some function that is only available in a console command. 
+Usually, you should refactor the command and move some logic into a service that can be 
+reused in the controller. However, when the command is part of a third-party library, you 
+wouldn't want to modify or duplicate their code, but want to directly execute the command 
+instead. 
+
+An example of this is sending the emails that Swift Mailer spooled earlier
+:doc:`using the swiftmailer:spool:send command </cookbook/email/spool>`. Symfony
+allows you to directly execute a registered ``Command`` inside your Controller::
+
+    // src/AcmeBundle/Controller/SpoolController.php
+    namespace AcmeBundle\Controller;
+
+    use Symfony\Component\Console\Output\StreamOutput;
+    use Symfony\Component\Console\Input\ArrayInput;
+    use Symfony\Bundle\FrameworkBundle\Console\Application;
+
+    class SpoolController
+    {
+        public function sendSpoolAction($messages=10)
+        {
+            $kernel = $this->get('kernel');
+            $application = new Application($kernel);
+            $application->setAutoExit(false);
+
+            $input = new ArrayInput(array(
+               'command' => 'swiftmailer:spool:send',
+               '--message-limit' => $messages,
+            ));
+            $output = new StreamOutput(tmpfile(), StreamOutput::VERBOSITY_NORMAL);
+            $application->run($input, $output);
+
+            rewind($output->getStream());
+            $content = stream_get_contents($output->getStream());
+            fclose($output->getStream());
+
+            return $content;
+        }
+    }
+
+Showing Colorized Command Output
+--------------------------------
+
+By telling the ``StreamOutput`` it is decorated via the third parameter, it will return 
+the Ansi color-coded content. The `SensioLabs AnsiToHtml converter`_ can be required 
+using ``Composer`` and helps you to get colorful html::
+
+    // src/AppBundle/Controller/SpoolController.php
+    namespace AppBundle\Controller;
+
+    use Symfony\Component\Console\Output\StreamOutput;
+    use Symfony\Component\Console\Input\ArrayInput;
+    use Symfony\Bundle\FrameworkBundle\Console\Application;
+    use SensioLabs\AnsiConverter\AnsiToHtmlConverter;
+
+    class SpoolController
+    {
+        public function sendSpoolAction($messages=10)
+        {
+            $kernel = $this->get('kernel');
+            $application = new Application($kernel);
+            $application->setAutoExit(false);
+
+            $input = new ArrayInput(array(
+               'command' => 'swiftmailer:spool:send',
+               '--message-limit' => $messages,
+            ));
+            $output = new StreamOutput(tmpfile(), StreamOutput::VERBOSITY_NORMAL, true);
+            $application->run($input, $output);
+
+            rewind($output->getStream());
+            $content = stream_get_contents($output->getStream());
+            fclose($output->getStream());
```

Comment:
Not sure if we should really show the complete code above. I think we can simply use the `// ...` placeholder here instead.